Source/WebCore/page/LocalFrame.cpp

```diff
@@ -1381,6 +1383,40 @@ void LocalFrame::setScrollingMode(ScrollbarMode scrollingMode)
         view->setCanHaveScrollbars(m_scrollingMode != ScrollbarMode::AlwaysOff);
 }
 
+#if ENABLE(CONTENT_EXTENSIONS)
+
+void LocalFrame::networkUsageDidExceedThreshold()
+{
+    ASSERT(!isMainFrame());
+
+    // If the frame has sticky user activation, don't do offloading.
+    if (RefPtr protectedWindow = window(); protectedWindow && protectedWindow->hasStickyActivation())
+        return;
+
+    showResourceMonitoringError("This frame was unloaded."_s);
```

Comment:
We should `RELEASE_LOG` something here so we know this happened when examining system logs.